Updates for `README.md` (#1010)
model-checking/kani: README.md

# Kani Rust Verifier

> The Kani Rust Verifier aims to be a bit-precise model checker for Rust.

Reviewer: ```suggestion
The Kani Rust Verifier is a bit-precise model checker for Rust.
```

Author: Done.

> Kani verifies memory safety, checks for different variants of undefined behavior, panics and user-defined assertions - even in unsafe Rust code.

Reviewer: A few suggestions:
- I think we should tie "unsafe Rust code" to memory safety because people assume that memory safety is checked by the compiler.
- Move checking of undefined behavior to the end.
Here is a suggested re-wording:
```suggestion
Kani verifies memory safety -- even in unsafe Rust code, verifies user-specified assertions as well as the absence of panics, and checks for some classes of undefined behavior.
```

Author: I liked your re-wording but at the same time thought it may be confusing. I made an attempt to convert this sentence into a bulleted list. How does it look to you?

## Installation

> Kani must be built from source. See [the installation guide](https://model-checking.github.io/kani/install-guide.html) for more details.

Reviewer: ```suggestion
Kani currently must be built from source. See [the installation guide](https://model-checking.github.io/kani/install-guide.html) for more details.
```

Author: Updated as "Kani must currently be built from source" because that's what we have in the install guide. If it's not ok, let me know and I will change both.
